src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -261,6 +261,11 @@ GcsTaskManager::GcsTaskManagerStorage::AddOrReplaceTaskEvent(
 
   // A new task event, add to storage and index.
 
+  // Bump the task counters by type.
+  if (events_by_task.has_task_info()) {
+    num_tasks_by_type_[events_by_task.task_info().type()]++;
```

Comment:
@rickyyx can you suggest if this is correct?

Reply:
Since we are technically tracking based on task attempt (task id + attempt number), if a task is run multiple times due to retry, this will be double counted.

Reply:
maybe we can only correct attempt number == 0?

Reply:
Ah, that's a good idea.